src/libs/data-utils.js

```diff
@@ -340,3 +356,231 @@ export const EntityUploader = class EntityUploader extends Component {
     ])])
   }
 }
+
+export const EntityRenamer = ({ entityType, entityName, workspaceId: { namespace, name }, onDismiss, onSuccess }) => {
+  const [newName, setNewName] = useState(entityName)
+  const [isBusy, setIsBusy] = useState()
+  const [takenName, setTakenName] = useState()
+
+  const doRename = async () => {
+    try {
+      setIsBusy(true)
+      await Ajax().Workspaces.workspace(namespace, name).renameEntity(entityType, entityName, _.trim(newName))
+      onSuccess()
+    } catch (e) {
```

Comment:
Can we use `withBusyState` here? I'm thinking maybe the `catch` is getting in the way somehow but I can't see the problem.
(Same question about `doEdit` below)

Reply:
So, since both the success case and the failure case end with the component unmounting, there's no need to remove the `isBusy` state, since it would just be a state transaction against an unmounting/unmounted component. The exception to this is the case where the name is already taken, but that's the only situation in which we really want to touch the busy state, since it's the only case where the component _won't_ unmount. For the entity editor, there isn't any case that shouldn't end in an unmount at all.